src/lib/cozy-client/slices/synchronization.js

```diff
@@ -75,23 +99,19 @@ export const syncDoctypeError = (doctype, error) => ({
   error
 })
 
-// TODO: not ideal because it will return false if no offline doctypes have been declared in the client config
-export const hasSyncStarted = state =>
-  Object.keys(state.cozy.synchronization).length !== 0
+export const hasSyncStarted = state => state.cozy.synchronization.started
```

Comment:
Not sure we want to expose external reducers organization in this internal reducer.
One simple example:

```js
// for illustration this reducer is dumb and returns a specific state
const children = (state, action) => [
  {
    name: 'Bob',
    age: 8
  },
  {
    name: 'Alice',
    age: 10
  }
]

// the children reducer is combined somewhere in the external organization
const parent = combineReducers({ children })

// Then come the selectors
const getAgeFromChildrenPointOfView = (state, index) => state[index].age

const getAgeFromParentPointOfView = (state, index) => state.children[index].age
```

Both selectors are ok, but not written in the same level. If you put the selector in the parent's reducer scope, it makes sens to take in consideration the state organization (`children` is a property of the `state`). But if you put the selector in the children's reducer scope, then, it doesn't make sense to illustrate the external state organization.

We should rather put selectors in different levels with every level responsible to access the sublevel selector:

```js
const getAgeFromChildrenPOV = (state, index) => state[index].age

const getAgeFromParentPOV = (state, index) => getAgeFromChilrenPOV(state.children, index)
```

Reply:
Right, I agree the `.cozy` here is a smell. We'll have to clean this up because we have this problem in other reducers, but it's outside the scope of this PR.

Reply:
I think the utility function `proxy` is useful :

Reply:
je ne vois pas bien l'avantage d'avoir
```js
const getAgeFromParentPOV = proxy(getAgeFromChildrenPOV, 'children')
```
plutôt que ça
```js
const getAgeFromParentPOV = (state, index) => getAgeFromChilrenPOV(state.children, index)
```

mis à part de se poser la question de ce que c'est que `proxy`.

Reply:
`getAgeFromParentPOV` n'est pas couplé à `state` dans ce cas.

Reply:
L'avantage n'est pas évident si on a seulement une méthode. Si on a plusieurs méthodes à proxier ca peut être intéressant.

```js
import customReducer, * as fromMyCustomReducer from 'reducers/custom'

export default combineReducers({
  custom: customReducer
})

const proxyAll = (methods, path) => mapValues(methods, method => proxy(method, path))

export { ...proxyAll(myCustomReducer, 'custom') } 
```